File: src/main/java/pl/training/workshop/m3/s10_isp/start/CashDesk.java
package pl.training.workshop.m3.s10_isp.start;

/** Klient: kasa. Używa sellTicket i refundTicket, a zależy od całego zaplecza. */
public final class CashDesk {
    private final CinemaAdminService backOffice;

    public CashDesk(CinemaAdminService backOffice) {
        this.backOffice = backOffice;
    }

    public String sell(String title, int seat) {
        return "bilet " + backOffice.sellTicket(title, seat) + ": " + title + ", miejsce " + seat;
    }

    public String refund(String ticketId) {
        return backOffice.refundTicket(ticketId);
    }
}

File: csharp/src/Training.Workshop/M3/S10Isp/Start/CashDesk.cs
namespace Training.Workshop.M3.S10Isp.Start;

/// <summary>Klient: kasa. Używa SellTicket i RefundTicket, a zależy od całego zaplecza.</summary>
public sealed class CashDesk
{
    private readonly ICinemaAdminService _backOffice;

    public CashDesk(ICinemaAdminService backOffice)
    {
        _backOffice = backOffice;
    }

    public string Sell(string title, int seat)
    {
        return "bilet " + _backOffice.SellTicket(title, seat) + ": " + title + ", miejsce " + seat;
    }

    public string Refund(string ticketId)
    {
        return _backOffice.RefundTicket(ticketId);
    }
}
